### PR TITLE
Initial work to refactor NSE

### PR DESCRIPTION
**Pull request by fkautz**

### CODE REVIEW

File: pkg/nsm/apis/nseconnect/nseconnect.proto [hunk 1]
```diff
@@ -5,66 +5,68 @@ package nseconnect;
...
 import "github.com/ligato/networkservicemesh/pkg/nsm/apis/common/common.proto";
 
-// EndpointConnectionRequest is sent by a NSM to NSE to build a connection.
-    message EndpointConnectionRequest {
-        string request_id = 1;
-        string network_service_name = 2;
-    }
-    
+// ConnectRequest is sent
+message ConnectRequest {
```
Comment (edwarnicke) [inline pkg/nsm/apis/nseconnect/nseconnect.proto]: Curious, why ConnectRequest not ConnectionRequest?

File: cmd/examples/nse/nse.go [hunk 2]
```diff
@@ -54,12 +55,13 @@ type nseConnection struct {
 	linuxNamespace     string
 }
 
-func (n nseConnection) RequestEndpointConnection(ctx context.Context, req *nseconnect.EndpointConnectionRequest) (*nseconnect.EndpointConnectionReply, error) {
+func (n nseConnection) Connect(ctx context.Context, req *nseconnect.ConnectRequest) (*nseconnect.ConnectReply, error) {
```
Comment (sbezverk) [inline cmd/examples/nse/nse.go]: This name does not provides any hints what this func is used for.

File: cmd/examples/nse/nse.go [hunk 3]
```diff
@@ -54,12 +55,13 @@ type nseConnection struct {
...
+	v4uuid := uuid.NewV4()
+	uuidString := v4uuid.String()
 
-	return &nseconnect.EndpointConnectionReply{
-		RequestId:          n.podUID,
-		NetworkServiceName: n.networkServiceName,
-		LinuxNamespace:     n.linuxNamespace,
+	return &nseconnect.ConnectReply{
+		RequestId: uuidString,
+		Mechanism: req.Mechanisms[0],
```
Comment (sbezverk) [inline cmd/examples/nse/nse.go]: what is the point of returning Mechanism that came in the request?

File: pkg/nsm/apis/nseconnect/nseconnect.proto [hunk 4]
```diff
@@ -5,66 +5,68 @@ package nseconnect;
...
+    string request_id = 1;
+    string network_service_name = 2;
+    map<string, string> context = 3;
+    repeated common.LocalMechanism mechanisms = 4;
```
Comment (sbezverk) [inline pkg/nsm/apis/nseconnect/nseconnect.proto]: Selection of LocalMechanism is done by NSM based on what client requested and what NSE advertised it supported for the specific network service. Then dataplane programs this selection and let know NSE which interface or whatever to use to serve NSE's network service. Not sure why exchanging LocalMechanism back and forth is needed.

**Comment by fkautz** [comment]
I should of posted earlier on this, don't merge. This was posted primarily to see how it breaks CI. It is nowhere near the final version.

**Pull request closed without merging**